File: app/backend/routes/health.rs
use axum::{
    response::sse::{Event, Sse},
    routing::get,
    Router,
    Json,
};
use std::time::Duration;
use futures_util::stream::Stream;
use tokio_stream::StreamExt;

pub fn router() -> Router {
    Router::new()
        .route("/", get(root))
        .route("/ping", get(ping))
}

async fn root() -> Json<serde_json::Value> {
    Json(serde_json::json!({ "message": "Welcome to AI Hedge Fund API" }))
}

async fn ping() -> Sse<impl Stream<Item = Result<Event, std::convert::Infallible>>> {
    let stream = tokio_stream::iter(0..5)
        .throttle(Duration::from_secs(1))
        .map(|i| {
            let data = serde_json::json!({
                "ping": format!("ping {}/5", i + 1),
                "timestamp": i + 1
            });
            let event = Event::default().data(data.to_string());
            Ok(event)
        });

    Sse::new(stream).keep_alive(axum::response::sse::KeepAlive::default())
}

File: app/backend/routes/health.py
from fastapi import APIRouter
from fastapi.responses import StreamingResponse
import asyncio
import json

router = APIRouter(prefix="/api")


@router.get("/health")
async def health():
    return {"status": "ok", "message": "AI Hedge Fund API"}


@router.get("/ping")
async def ping():
    async def event_generator():
        for i in range(5):
            data = {"ping": f"ping {i+1}/5", "timestamp": i + 1}
            yield f"data: {json.dumps(data)}\n\n"
            await asyncio.sleep(1)

    return StreamingResponse(event_generator(), media_type="text/event-stream")
